### PR TITLE
Prefactor: add declarative config health API

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

This PR is the first PR in a series of PRs to switch the usage of the integration health datastore to the newly declarative configuration health datastore within the declarative configuration manager and the UI components.

The first PR is a prefactor and creates a public API for the UI to consume to retrieve the declarative configuration health status.

In a follow-up PR, finally the following will be done:
- the manager itself will switch from integration health to declarative config health.
- E2E tests will be adjusted to verify the new API instead of the previous one.
- the UI will ingest the data from the declarative config health API instead of the integration health API within the system health dashboard.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

- see CI passing.

### CODE REVIEW

**Comment by dhaus67** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6304** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6304" target="_blank">Graphite</a> 
    * **PR #6316** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6316" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/6316?utm_source=stack-comment).

File: central/declarativeconfig/health/service/singleton.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import (
+	"github.com/stackrox/rox/central/declarativeconfig/health/datastore"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+var (
+	once sync.Once
+
+	ds Service
```
Comment (ivan-degtiarenko) [inline central/declarativeconfig/health/service/singleton.go:11]: nit: not sure `ds` applies here

File: proto/api/v1/declarative_config_health_service.proto [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+service DeclarativeConfigHealthService {
+
+    rpc GetDeclarativeConfigHealths (Empty) returns (GetDeclarativeConfigHealthsResponse) {
+        option (google.api.http) = {
+            get: "/v1/declarative-config/health"
+        };
+    }
+
+    rpc GetDeclarativeConfigHealth (ResourceByID) returns (GetDeclarativeConfigHealthResponse) {
```
Comment (ivan-degtiarenko) [inline proto/api/v1/declarative_config_health_service.proto:29]: I think that:
1. Considering this API is unused, should we even expose it?
2. If we are to expose it, let's explain how the API client can obtain `id` for config health (e.g. how they should transform the name of a resource to UUID using a certain algorithm)

Comment (dhaus67) [reply proto/api/v1/declarative_config_health_service.proto:29]: Yeah I'm fine removing it tbh 👍

**Review by ivan-degtiarenko — APPROVED** [review]
LGTM 👍

**Pull request merged**